zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/appliers/MappingAppliersTest.java

```diff
@@ -50,12 +51,70 @@ public void setup() {
     tenantState = processingState.getTenantState();
     authorizationState = processingState.getAuthorizationState();
     groupState = processingState.getGroupState();
-    mappingDeletedApplier = new MappingDeletedApplier(processingState);
+    mappingDeletedApplier = new MappingDeletedApplier(processingState.getMappingState());
+    mappingUpdatedApplier = new MappingUpdatedApplier(processingState.getMappingState());
   }
 
   @Test
   void shouldDeleteMapping() {
     // given
+    final var mappingRecord = createMapping();
+
+    // when
+    mappingDeletedApplier.applyState(mappingRecord.getMappingKey(), mappingRecord);
+
+    // then
+    assertThat(mappingState.get(mappingRecord.getMappingKey())).isEmpty();
+  }
+
+  @Test
+  public void shouldThrowExceptionWhenDeleteNotExistingMapping() {
+    // given
+    final long mappingKey = 1L;
+    final var mappingRecord = new MappingRecord().setMappingKey(mappingKey);
+
+    // when + then
+    assertThatThrownBy(() -> mappingDeletedApplier.applyState(mappingKey, mappingRecord))
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessageContaining(
+            "Expected to delete mapping with key '1', but a mapping with this key does not exist.");
+  }
+
+  @Test
+  void shouldUpdateMapping() {
+    // given
+    final var mappingRecord = createMapping();
```

Comment:
🔧 Can we ensure we use unique keys/ids here? If we reuse them we run the risk of the test cases not being isolated.